packages/standard-library.md

```diff
@@ -5,3 +5,173 @@ The Pony standard library is a collection of packages that can each be used as n
 There is also a special package in the standard library called __builtin__. This contains various types that the compiler has to treat specially and are so common that all Pony code needs to know about them. All Pony source files have an implicit `use "builtin"` command. This means all the types defined in the package builtin are automatically available in the type namespace of all Pony source files.
 
 Documentation for the standard library is [available online](http://stdlib.ponylang.org/)
+
+## A quick glance at the standard library
+
+There is some package in the standard library that are so useful that deserve to be presented here.
+
+### The debug package
+
+The debug package is useful when you want to do print something without having to pass the ```Env``` around. To use the debug package just import and use it this way:
+
+```pony
+actor Main
+  new create(env: Env) =>
+    Debug.out("This will only bee seen when configured for debug info")
+```
+
+Note that for the debug statement to show up you will need to compile your program with the ```-d``` option.
+
+
+### The itertools package
+
+The itertools package is a really useful package when dealing with collections. If the words filter, map, fold and zip speak to you then take a look at this package. 
+Here is a quick example that from start with an array of numbers, add one to each, then only keep the even ones and finally print each value.
+
+```pony
+use "itertools"
+
+actor Main
+  new create(env: Env) =>
+    Iter[I64]([1; 2; 3; 4; 5].values())
+      .map[I64]({(x) => x + 1 })
+      .filter({(x) => (x % 2) == 0 })
+      .map[None]({(x) => env.out.print(x.string()) })
+      .run()
+```
+
+### The collections package
+
+Speaking about collections it is worth taking a look at the package of the same name. The collections package contains the generic common containers such as List, HasMap, HashSet.
+
+```pony
+use "collections"
+
+actor Main
+  new create(env: Env) =>
+    var a_list = List[I64].from([1; 2; 3; 4; 5])
+    a_list.push(6)
+    for elem in a_list.values() do
+      env.out.print(elem.string())
+    end
+```
+
+```pony
+use "collections"
+
+primitive StringHasher
+  fun hash(str : String) : U64 =>
+    str.hash()
+
+  fun eq(str1 : String, str2 : String) : Bool =>
+    str1 == str2
+
+actor Main
+  new create(env: Env) =>
+    var inhabitants_count_per_country = HashMap[String, F32, StringHasher]
+    
+    inhabitants_count_per_country.update("France" , F32(70_000_000))
+    inhabitants_count_per_country.update("USA" , F32(323_000_000))
+    
+    for (key, value) in inhabitants_count_per_country.pairs() do
+      env.out.print(key + " has " + value.string() + " inhabitants")
+    end
+```
+
```

Comment:
Maybe also at least mention the `collections.persistent` package.

Reply:
Just to make sure, the difference between the 2 packages are that when you do an operation on a collection from collections.persistent it is making a copy of the old and return a new one with the operation applied?